### PR TITLE
[WIP] examples with opencv4/origami

### PR DESCRIPTION
**Pull request by hellonico**
## Description ##
This PR uses OpenCV/Origami as the image manipulation library.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [x ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
Basically using library: [origami "4.0.0-2"] with imports: 
```
[opencv4.core :as cv]
[opencv4.utils :as cvu]
```

Whenever possible.

### CODE REVIEW

File: contrib/clojure-package/examples/gan/src/gan/viz.clj [hunk 1]
```diff
@@ -37,29 +32,11 @@
...
-              (let [result (new Mat height width (CvType/CV_8U))
-                    _ (.put result (int 0) (int 0) (byte-array raw-data))]
-                result))]
-    (do
-      (if flip
-        (let [result (new Mat)
-              _ (Core/flip img result (int 0))]
-          result)
-        img))))
+              (cv/>> (new-mat height width CV_8UC1) (byte-array raw-data)))]
```
Comment (gigasquid) [inline contrib/clojure-package/examples/gan/src/gan/viz.clj]: neat threading :)

File: contrib/clojure-package/examples/neural-style/src/neural_style/core.clj [hunk 2]
```diff
@@ -41,63 +40,63 @@
...
+;;;;
+
+(defn image->ndarray 
+  "normalize the pixels for vgg19"
+  [simg]
+  (let [h (.height simg) w (.width simg)]
+    (println "The nd image size is:" {:height h :width w})
+    (-> simg 
+      (cv/convert-to! cv/CV_8SC3 0.5) 
+      (cv/add! (cv/new-scalar -103.939 -116.779 -123.68) )
```
Comment (gigasquid) [inline contrib/clojure-package/examples/neural-style/src/neural_style/core.clj]: Cool - this is a nice way to do this

File: contrib/clojure-package/examples/neural-style/src/neural_style/core.clj [hunk 3]
```diff
@@ -41,63 +40,63 @@
...
+; IMAGE MANIPULATION
+;;;;
+
+(defn image->ndarray 
+  "normalize the pixels for vgg19"
+  [simg]
+  (let [h (.height simg) w (.width simg)]
+    (println "The nd image size is:" {:height h :width w})
+    (-> simg 
+      (cv/convert-to! cv/CV_8SC3 0.5)
```
Comment (gigasquid) [inline contrib/clojure-package/examples/neural-style/src/neural_style/core.clj]: what does the `0.5` in `convert-to!` do?

File: contrib/clojure-package/examples/pre-trained-models/src/pre_trained_models/predict_image.clj [hunk 4]
```diff
@@ -40,21 +40,12 @@
...
-        image (-> (img/load-image fname)
-                  (img/resize h w))
-        pixels (img/get-pixels image)
-        rgb-pixels (reduce (fn [result pixel]
-                             (let [[rs gs bs] result
-                                   [r g b _] (pixel/unpack-pixel pixel)]
-                               [(conj rs r) (conj gs g) (conj bs b)]))
-                           [[] [] []]
-                           pixels)]
-    (when show? (img/show image))
```
Comment (gigasquid) [inline contrib/clojure-package/examples/pre-trained-models/src/pre_trained_models/predict_image.clj]: Is there a way to show the image to the user if the `show?` flag is true? `imshow`?

File: contrib/clojure-package/examples/infer/objectdetector/src/infer/draw.clj [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+;; limitations under the License.
+;;
+
+(ns infer.draw
+	(:require 
+		[opencv4.colors.rgb :as rgb]
+		[opencv4.core :refer [FONT_HERSHEY_PLAIN imread imwrite new-point put-text! rectangle]]))
+
+(defn black-boxes! [img results]
+    (doseq [
```
Comment (gigasquid) [inline contrib/clojure-package/examples/infer/objectdetector/src/infer/draw.clj]: I feel like the formatting is a bit off in this function - I know if varies from editor to editor.  There is a tool to keep the style consistent between contributors, (I should run this more often I know :) )

File: contrib/clojure-package/examples/gan/src/gan/viz.clj [hunk 6]
```diff
@@ -18,14 +18,9 @@
 (ns gan.viz
```
Comment (gigasquid) [inline contrib/clojure-package/examples/gan/src/gan/viz.clj]: I ran the GAN example and everything looks great 💯  - I even think it's faster 🚗 
Nice job.

File: contrib/clojure-package/examples/neural-style/project.clj [hunk 7]
```diff
@@ -18,8 +18,8 @@
 (defproject neural-style "0.1.0-SNAPSHOT"
```
Comment (gigasquid) [inline contrib/clojure-package/examples/neural-style/project.clj]: Ran the example - works great!

File: contrib/clojure-package/examples/infer/objectdetector/project.clj [hunk 8]
```diff
@@ -17,9 +17,12 @@
 
 (defproject objectdetector "0.1.0-SNAPSHOT"
   :description "Object detection using infer with MXNet"
+  :repositories [["vendredi" "https://repository.hellonico.info/repository/hellonico/"]]
   :plugins [[lein-cljfmt "0.5.7"]]
+  :aliases {"sample-detector" ["run" "--" "-m" "models/resnet50_ssd/resnet50_ssd_model" "-i" "images/dog.jpg" "-d" "images/"]}
```
Comment (gigasquid) [inline contrib/clojure-package/examples/infer/objectdetector/project.clj]: nice alias - can you please add to the README too?

File: contrib/clojure-package/examples/infer/objectdetector/src/infer/objectdetector_example.clj [hunk 9]
```diff
@@ -106,10 +127,9 @@
                   factory
                   {:contexts [(context/default-context)]})]
     (println "Object detection on a single image")
-    (print-predictions (detect-single-image detector input-image) width height)
```
Comment (gigasquid) [inline contrib/clojure-package/examples/infer/objectdetector/src/infer/objectdetector_example.clj]: Could we also leave the printing of the predictions to the console in as well?

File: contrib/clojure-package/examples/infer/objectdetector/src/infer/objectdetector_example.clj [hunk 10]
```diff
@@ -63,19 +68,34 @@
...
+        topk 5]
+        (process-results 
+        [input-image]
+        (infer/detect-objects detector image topk)
+        output-dir)))
 
 (defn detect-images-in-dir
   "Detect objects in all jpg images in the directory"
-  [detector input-dir]
+  [detector input-dir output-dir]
```
Comment (gigasquid) [inline contrib/clojure-package/examples/infer/objectdetector/src/infer/objectdetector_example.clj]: Could we make the default output-dir for the example `results` or `image-results`?

File: contrib/clojure-package/examples/infer/objectdetector/src/infer/objectdetector_example.clj [hunk 11]
```diff
@@ -20,6 +20,7 @@
             [org.apache.clojure-mxnet.infer :as infer]
             [org.apache.clojure-mxnet.layout :as layout]
             [clojure.java.io :as io]
+            [infer.draw :as draw]
```
Comment (gigasquid) [inline contrib/clojure-package/examples/infer/objectdetector/src/infer/objectdetector_example.clj]: I ran the example and the drawing of the bounding boxes is fantastic!

Can you please merge this PR with master because the output of the predictions has now changed shape a bit 
https://github.com/apache/incubator-mxnet/pull/13864

Object detection now returns a map
`{:class car, :prob 0.99847263, :x-min 0.6097917, :y-min 0.1406818, :x-max 0.8906532, :y-max 0.29426125}`

**Comment by gigasquid** [comment]
Thanks for addressing the feedback so quickly @hellonico 😄 
I plan to take one more test drive through it shortly 🏎
